gdx/src/com/badlogic/gdx/graphics/g2d/Sprite.java

```diff
@@ -338,45 +339,50 @@ public void translate (float xAmount, float yAmount) {
 	/** Sets the color used to tint this sprite. Default is {@link Color#WHITE}. */
 	public void setColor (Color tint) {
 		color.set(tint);
-		float color = tint.toFloatBits();
+		packedColor = tint.toFloatBits();
 		float[] vertices = this.vertices;
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		vertices[C1] = packedColor;
+		vertices[C2] = packedColor;
+		vertices[C3] = packedColor;
+		vertices[C4] = packedColor;
 	}
 
 	/** Sets the alpha portion of the color used to tint this sprite. */
 	public void setAlpha (float a) {
-		color.a = a;
-		float color = this.color.toFloatBits();
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		if (color.a != a) {
+			color.a = a;
+			packedColor = color.toFloatBits();
+			vertices[C1] = packedColor;
+			vertices[C2] = packedColor;
+			vertices[C3] = packedColor;
+			vertices[C4] = packedColor;
+		}
 	}
 
 	/** @see #setColor(Color) */
 	public void setColor (float r, float g, float b, float a) {
 		color.set(r, g, b, a);
-		float color = this.color.toFloatBits();
+		packedColor = color.toFloatBits();
 		float[] vertices = this.vertices;
-		vertices[C1] = color;
-		vertices[C2] = color;
-		vertices[C3] = color;
-		vertices[C4] = color;
+		vertices[C1] = packedColor;
+		vertices[C2] = packedColor;
+		vertices[C3] = packedColor;
+		vertices[C4] = packedColor;
 	}
 
 	/** Sets the color of this sprite, expanding the alpha from 0-254 to 0-255.
 	 * @see #setColor(Color)
 	 * @see Color#toFloatBits() */
 	public void setPackedColor (float packedColor) {
-		Color.abgr8888ToColor(color, packedColor);
-		float[] vertices = this.vertices;
-		vertices[C1] = packedColor;
-		vertices[C2] = packedColor;
-		vertices[C3] = packedColor;
-		vertices[C4] = packedColor;
+		if (this.packedColor != packedColor) {
```

Comment:
hehehe, would adding an `abs()` fix all cases you can think of?

Reply:
If by "fix all cases" you mean "cause more bugs" then sure. Calling abs() on any float (except NaN?) will just zero out the sign bit, making the number positive. Comparing packed floats made from `0xBF000000` and `0x3F000000` currently considers them different, as it should. If both had abs() called on them and they were then compared, they would compare as equal, which is probably not what you want. As far as I can tell, the only correct way to do a bit-by-bit comparison of floats is to convert them to bits, in our case with either `NumberUtils.floatToIntColor(packedColor)` or the probably-faster `(NumberUtils.floatToIntBits(packedColor) & 0xFEFFFFFF)` (remember the parentheses around the `&` operation!), and then compare the resulting ints.

Reply:
I mean adding an abs() in the case == 0. Or are there more cases you are aware in which this issue can happen in addition to 0?

Reply:
I'm not sure what `Math.abs(-0f)` would accomplish either! It would still treat -0f and 0f as equal when they aren't. You can look at the hashCode() method in Color to see how it uses the bits of each float to compare -0f and 0f as different values. If you're concerned about the speed of the method that does the work, `Float.floatToIntBits()`, on HotSpot it's an intrinsic and I believe it is also on ART for Android. The body of the intrinsic is C++ code that uses a union type to get the bits from a float using only two assignment operations, if I remember it right.

Reply:
Ok, I'm not really concerned as whatever the check is it's only going to be executed if ==0.  What I'm trying to determine here is if adding code that handles the special case of dealing with 0/-0 we fix the problem or not.

Reply:
The main check I've come up with so far is:
```java
(packedColor != this.packedColor || (packedColor  == 0f && this.packedColor == 0f && 1f/packedColor  != 1f/this.packedColor))
```
It's pretty nasty, BUT it only executes in this corner case. It takes advantage of the property that 1f/0f is positive infinity and 1f/-0f is negative infinity. There may be a better non-bitwise alternative to handling -0f, but I still think you'd be have a good option with storing the new `packedColor`'s bits, comparing them to `this.packedColor`'s bits, and then reusing the stored bits inside the code for `Color.abgr8888ToColor()` (inlining it to do so). The check with two divisions, though, does only execute when both sides are equal to 0, which might make it cost very little with branch prediction.

Reply:
Ok, so we just have to deal with 0. Even if your check is a bit more compact I think I prefer your previous suggestion of using `Float.floatToIntBits`.
`packedColor != this.packedColor || (packedColor  == 0f && this.packedColor == 0f && Float.floatToIntBits(packedColor) != Float.floatToIntBits(this.packedColor))`